modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ITMetaStorageServicePersistenceTest.java

```diff
@@ -0,0 +1,465 @@
+    ) throws ExecutionException, InterruptedException {
+        Entry entry = metaStorage.get(key).get();
+
+        assertEquals(isTombstone, entry.tombstone());
+
+        assertEquals(isEmpty, entry.empty());
+
+        assertArrayEquals(
+            value,
+            entry.value()
+        );
+
+        assertEquals(revision, entry.revision());
+
+        assertEquals(updateCounter, entry.updateCounter());
+    }
+
+    /** */
+    @SuppressWarnings("BusyWait") public static boolean waitForCondition(BooleanSupplier cond, long timeout) {
+        long stop = System.currentTimeMillis() + timeout;
+
+        while (System.currentTimeMillis() < stop) {
+            if (cond.getAsBoolean())
+                return true;
+
+            try {
+                sleep(50);
+            }
+            catch (InterruptedException e) {
+                return false;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param cluster The cluster.
+     * @param exp Expected count.
+     * @param timeout The timeout in millis.
+     * @return {@code True} if topology size is equal to expected.
+     */
+    private boolean waitForTopology(ClusterService cluster, int exp, int timeout) {
+        return waitForCondition(() -> cluster.topologyService().allMembers().size() >= exp, timeout);
+    }
+
+    /**
+     * @return Local address.
+     */
+    public static String getLocalAddress() {
+        try {
+            return InetAddress.getLocalHost().getHostAddress();
+        }
+        catch (UnknownHostException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Creates a cluster service.
+     */
+    protected ClusterService clusterService(String name, int port, List<NetworkAddress> servers) {
+        var context = new ClusterLocalConfiguration(name, port, servers, SERIALIZATION_REGISTRY);
+
+        var network = NETWORK_FACTORY.createClusterService(context);
+
+        network.start();
+
+        cluster.add(network);
+
+        return network;
+    }
+
+    /**
+     * Starts a raft server.
+     *
+     * @param idx Server index (affects port of the server).
+     * @param clo Server initializer closure.
+     * @return Server.
+     * @throws IOException If failed.
+     */
+    private JRaftServerImpl startServer(int idx, Consumer<RaftServer> clo) throws IOException {
+        var addr = new NetworkAddress(getLocalAddress(), PORT);
+
+        ClusterService service = clusterService("server" + idx, PORT + idx, List.of(addr));
+
+        Path jraft = workDir.resolve("jraft" + idx);
+
+        JRaftServerImpl server = new JRaftServerImpl(service, jraft.toString(), FACTORY) {
+            @Override public void shutdown() throws Exception {
+                super.shutdown();
+
+                service.shutdown();
+            }
+        };
+
+        clo.accept(server);
+
+        servers.add(server);
+
+        assertTrue(waitForTopology(service, servers.size(), 3_000));
```

Comment:
do we need to wait for topology on every server start? Can we start a bunch of servers and then wait?